### PR TITLE
ts_state_t refactor

### PR DESCRIPTION
**Pull request by melouber**
* added comment about possible states
* renamed `LOCKED` to `BLOCKED` (as now we have `td_blockedq`, `TDS_BLOCKED`, `ts_blocked`)
* added assert that `ts_for_td->ts_wchan == NULL` when `ts_for_td` is in `FREE_BLOCKED` state

### CODE REVIEW

File: sys/turnstile.c [hunk 1]
```diff
@@ -14,7 +14,26 @@
...
+ *   > ts formerly owned by a thread td
+ *   > td is now blocked on mutex mtx
+ *   > td was not the first one to block on mtx
+ *   > ts->ts_wchan is equal to NULL
+ *
+ * - USED_BLOCKED:
+ *   > ts formerly owned by a thread td
+ *   > td is now blocked on mutex mtx
+ *   > td was the first one to block on mtx
+ *   > ts->ts_wchan is not equal to NULL
```
Comment (ilikeheaps) [inline sys/turnstile.c]: Isn't `ts->ts_wchan` equal to the address of `mtx`?

Comment (melouber) [reply sys/turnstile.c]: It is. But I didn't know whether we want such detailed information. Do we?

Comment (ilikeheaps) [reply sys/turnstile.c]: I feel more secure when I know what value a variable has instead of knowing what value it doesn't have.

Comment (melouber) [reply sys/turnstile.c]: Done.

**Pull request merged**